cmd/cosign/cli/generate.go

```diff
@@ -59,18 +60,18 @@ EXAMPLES
 			if len(args) != 1 {
 				return flag.ErrHelp
 			}
-			return GenerateCmd(ctx, args[0], annotations.annotations, os.Stdout)
+			return GenerateCmd(ctx, regOpts, args[0], annotations.annotations, os.Stdout)
 		},
 	}
 }
 
-func GenerateCmd(_ context.Context, imageRef string, annotations map[string]interface{}, w io.Writer) error {
+func GenerateCmd(ctx context.Context, regOpts RegistryOpts, imageRef string, annotations map[string]interface{}, w io.Writer) error {
 	ref, err := name.ParseReference(imageRef)
 	if err != nil {
 		return err
 	}
 
-	get, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	get, err := remote.Get(ref, regOpts.GetRegistryClientOpts(ctx)...)
```

Comment:
Is dropping the LHS intentional, since IIRC it's the default behavior?

Reply:
?

The effect of the change on this line specifically should be to a) propagate ctx, b) allow for broken TLS when the flag is set

Reply:
I think what confused me was that the LHS wasn't using `DefaultRegistryClientOpts` and the implementation of this function delegates to that which hides the fact that it is a superset.  I left a comment on that below, let's kill `DefaultRegistryClientOpts` completely if we can.

Reply:
roger